### PR TITLE
feat: port report-db command from 1.x

### LINKED ISSUE #23892: Modify influx_inspect to compute Cloud 2 cardinality [port to master]
Port https://github.com/influxdata/influxdb/issues/23356 to master
PR is https://github.com/influxdata/influxdb/pull/23351

### PR DESCRIPTION
**Pull request by jeffreyssmith2nd**
- Closes #23892 

### Severity (delete section if not relevant)
 i.e., ("recommend to upgrade immediately", "upgrade at your leisure", etc.)

### Note for reviewers:
Check the semantic commit type:
 - Feat: a feature with user-visible changes
 - Fix: a bug fix that we might tell a user “upgrade to get this fix for your issue”
 - Chore: version bumps, internal doc (e.g. README) changes, code comment updates, code formatting fixes… must not be user facing (except dependency version changes)
 - Build: build script changes, CI config changes, build tool updates
 - Refactor: non-user-visible refactoring
 - Check the PR title: we should be able to put this as a one-liner in the release notes

### CODE REVIEW

File: cmd/influxd/inspect/report_db/report_db.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+			Default: false,
+		},
+		{
+			DestP:   &flags.exact,
+			Flag:    "exact",
+			Desc:    "report exact counts",
+			Default: false,
+		},
+		{
+			DestP:   &flags.rollup,
```
Comment (davidby-influx) [inline cmd/influxd/inspect/report_db/report_db.go:77]: Does this work for 2.0 with buckets and stuff?  Should the arguments to rollup level be different?

Comment (davidby-influx) [reply cmd/influxd/inspect/report_db/report_db.go:77]: Is the structure org/bucket/measurement?

Comment (davidby-influx) [reply cmd/influxd/inspect/report_db/report_db.go:77]: Or are orgs not in the directory structure?  

https://docs.influxdata.com/influxdb/v2.5/reference/internals/file-system-layout/?t=Linux#tsm-directories-and-files-layout

**Review by davidby-influx — CHANGES_REQUESTED** [review]
I think we need to change the arguments to reflect buckets and possibly orgs as opposed to databases and retention policies.

File: cmd/influxd/inspect/report_db/report_db.go (hunk 1 above)
Comment (jeffreyssmith2nd) [reply cmd/influxd/inspect/report_db/report_db.go:77]: It does work for 2.0, though I did need to rename a few things to have it make sense for 2.0 (pushed a new commit for that).

1.x: `db_name/rp_name/shard_num/tsm_file`
2.x: `bucket_name/rp_name/shard_num/tsm_file`

Screenshot 2022-11-18 at 08 39 54

**Review by davidby-influx — APPROVED** [review]
LGTM

**Pull request merged**